pulsar-broker/src/test/java/org/apache/pulsar/broker/service/persistent/PersistentTopicTest.java

```diff
@@ -271,4 +272,70 @@ public void testPersistentPartitionedTopicUnload() throws Exception {
             producer.close();
         }
     }
+
+    @Test
+    public void testCreateSchemaAfterDeletion() throws Exception {
+        //init namespace
+        final String myNamespace = "prop/ns";
+        admin.namespaces().createNamespace(myNamespace, Sets.newHashSet("test"));
+        final String topicName = "persistent://prop/ns/test-create-schema-after-deletion" + UUID.randomUUID();
+
+        // create namespace
+        // Create a topic with `Person`
+        try (Producer<Person> producer = pulsarClient.newProducer(Schema.AVRO(Person.class))
+                .topic(topicName)
+                .create()
+        ) {
+            Person person = new Person();
+            person.setName("Tom Hanks");
+            person.setAge(60);
+
+            producer.send(person);
+
+        }
+
+        // delete the topic
+        admin.topics().delete(topicName);
+
+        try (Producer<Student> ignored = pulsarClient.newProducer(Schema.AVRO(Student.class))
+                .topic(topicName)
+                .create()) {
+            Assert.fail("Should fail to create a the producer with a new schema since the schema is not deleted.");
+        } catch (PulsarClientException pce) {
+            Assert.assertTrue(pce instanceof PulsarClientException.IncompatibleSchemaException);
+        }
+
+        // delete the schema
+        admin.schemas().deleteSchema(topicName);
+
+        // after deleting the schema, try to create a topic with a different schema
+        try (Producer<Student> producer = pulsarClient.newProducer(Schema.AVRO(Student.class))
+                .topic(topicName)
+                .create()
+        ) {
+            Student student = new Student();
+            student.setName("Tom Jerry");
+            student.setAge(30);
+            student.setGpa(6);
+            student.setGpa(10);
```

Comment:
Is there any reason to invoke `setGpa` twice?

Reply:
fixed